canonical-websites/discourse-docs
# Dynamic navigation support

Quite a few changes:

- Create a basic model for requesting data from Discourse
- Use the exact URL structure of Discourse (so we don't need to transform the nav markup)
- Redirect `/` to the "frontpage" (in this case, "documentation outline")
- Pull the navigation from the "frontpage"
- If we get given a redirect by Discourse, send the equivalent redirect to the user

Still to do (create issues):

- [ ] Handle errors in the navigation / frontpage structure elegantly
- [ ] Update the styling to match the new styling on docs.snapcraft.io
- [ ] Decide what to do about multiple `<h1>`s
- [x] Add a link to the forum post on each page
- [ ] Test the cache

QA
--

``` bash
./run
```

Go to http://0.0.0.0:8029, check you get redirected to http://0.0.0.0:8029/t/documentation-outline/3781, and that you see the content and navigation from https://forum.snapcraft.io/t/documentation-outline/3781.

Browse around, check the other pages you find also work and reflect the content on https://forum.snapcraft.io.
